xtask/src/main.rs

```diff
@@ -39,6 +41,37 @@ fn test(params: test::Params) -> anyhow::Result<()> {
     Ok(())
 }
 
+// Set up environment variables needed by the compilation and testing process.
+//
+// If ci flag is specified, all the binaries are added to the path,
+// and also the path to the sovereign constant manifest is added to the env variables.
+fn init_env(ci: bool) -> anyhow::Result<()> {
+    if ci {
+        #[rustfmt::skip]
+        let project_dir = duct::cmd!(
+            "sh", "-c",
+            "cargo locate-project | jq -r '.root' | grep -oE '^.*/'"
+        )
+        .stdout_capture()
+        .run()?;
+        let project_dir = std::str::from_utf8(&project_dir.stdout)?.trim();
+
+        let path = std::env::var("PATH").unwrap_or_else(|_| "".to_string());
+
+        // To ensure persistent storage between runs,
+        // all cargo binaries are compiled in the following folder in ci
+        let new_path = format!("/cargo_target/release/:{}", path);
+        std::env::set_var("PATH", new_path);
+
+        std::env::set_var(
+            "CONSTANTS_MANIFEST",
+            format!("{}demo/sovereign/constants.json", project_dir),
```

Comment:
For how I made the regex it will contain the '\' but you pointed out something really interesting and using Path it becomes a lot cleaner and less error prone:

```rust
        let path = std::path::Path::new(project_dir).join("demo/sovereign/constants.json");
        if !path.exists() {
            bail!(
                "The `constants.json` file for Sovereign does not exist,\n \
                   or it is not in the expected position, `demo/sovereign/constants.json`"
            )
        }
        std::env::set_var("CONSTANTS_MANIFEST", path);

```